diff -ur a/nixos/modules/security/pam.nix b/nixos/modules/security/pam.nix
--- a/nixos/modules/security/pam.nix	2020-10-19 03:13:06.232039662 +0100
+++ b/nixos/modules/security/pam.nix	2020-10-19 03:22:37.412405254 +0100
@@ -893,7 +893,7 @@
         "mr ${pkgs.ecryptfs}/lib/security/pam_ecryptfs.so,"}
       ${optionalString (isEnabled (cfg: cfg.pamMount))
         "mr ${pkgs.pam_mount}/lib/security/pam_mount.so,"}
-      ${optionalString config.services.samba.syncPasswordsByPam
+      ${optionalString config.services.samba.enable
         "mr ${pkgs.samba}/lib/security/pam_smbpass.so,"}
       ${optionalString (isEnabled (cfg: cfg.enableGnomeKeyring))
         "mr ${pkgs.gnome3.gnome-keyring}/lib/security/pam_gnome_keyring.so,"}
