diff --git a/server.go b/server.go
index 261bfce..469129a 100644
--- a/server.go
+++ b/server.go
@@ -33,6 +33,9 @@ type Config struct {
 	DbDriver string
 	DbDsn    string
 	Listen   string
+	TLS      bool
+	CertFile string
+	KeyFile  string
 
 	AccountCacheTtl time.Duration
 	TweetCacheTtl   time.Duration
@@ -881,5 +884,9 @@ func main() {
 	})
 
 	log.Printf("listening on: %s", inst.Config.Listen)
-	http.ListenAndServe(inst.Config.Listen, Log(r))
+	if inst.Config.TLS {
+		http.ListenAndServeTLS(inst.Config.Listen, inst.Config.CertFile, inst.Config.KeyFile, Log(r))
+	} else {
+		http.ListenAndServe(inst.Config.Listen, Log(r))
+	}
 }
