--- a/linux-user/elfload.c	2016-09-02 12:34:22.000000000 -0300
+++ b/linux-user/elfload.c	2017-07-09 18:44:22.420244038 -0300
@@ -1419,7 +1419,7 @@
  * dependent on stack size, but guarantee at least 32 pages for
  * backwards compatibility.
  */
-#define STACK_LOWER_LIMIT (32 * TARGET_PAGE_SIZE)
+#define STACK_LOWER_LIMIT (128 * TARGET_PAGE_SIZE)
 
 static abi_ulong setup_arg_pages(struct linux_binprm *bprm,
                                  struct image_info *info)
